tests/pathing.py

```diff
@@ -0,0 +1,517 @@
+
+        Parameters
+        ----------
+            roll : float
+                Roll angle to be passed to drone given in radians
+            pitch : float
+                Pitch angle to be passed to drone given in radians
+            yaw : float
+                Yaw angle to be passed to drone given in radians
+            z : float
+                A z (height) value to be passed to the drone given relative to the ground
+                Drone automatically travels to this position, smoothing its movement automatically
+            duration : float
+                Duration for the inputs to be passed to the drone given in seconds
+
+        Notes
+        -----
+            (from airsimdroneracinglab documenation)
+            Roll angle, pitch angle, and yaw angle are given in radians, in the body frame.
+            The body frame follows the Front Left Up (FLU) convention, and right-handedness.
+        """
+        self.airsim_client.moveByRollPitchYawZAsync(
+            roll, pitch, yaw, z, duration, vehicle_name=self.drone_name
+        )
+
+    def get_gate_pose(self, gate_index: int) -> airsimdroneracinglab.Pose:
+        """
+        Gets a list of gates from the airsim client,
+        then returns the pose of the gate requested in the form of an object.
+
+        Paremeters
+        ----------
+            gate_index : int
+                Index of requested/next gate (starts at 0)
+
+        Returns
+        -------
+            gate_position : airsimdroneracinglab.Pose
+                Position of the next gate given as an object containing a Quaternionr and Vector3r
+
+        """
+        gate_list: list[str] = self.airsim_client.simListSceneObjects("Gate.*")
+        gate_position: airsimdroneracinglab.Pose = self.airsim_client.simGetObjectPose(
+            gate_list[gate_index]
+        )
+        return gate_position
+
+    def get_drone_pose(self) -> airsimdroneracinglab.Pose:
+        """
+        Gets the current position of the drone from the airsim client and returns it as an object.
+
+        Returns
+        -------
+            drone_position : airsimdroneracinglab.Pose
+                Current position of drone given as an object containing a Quaternionr and Vector3r
+        """
+        drone_position: airsimdroneracinglab.Pose = self.airsim_client.simGetObjectPose(
+            self.drone_name
+        )
+        return drone_position
+
+    def get_drone_y_velocity(self) -> float:
+        """
+        Calculates the drone's Y velocity using the difference in it's position over
+        a fixed interval of time.
+
+        Returns
+        -------
+            y_velocity : float
+                Drone's current y velocity (perpendicular left-right velocity) in relation to frame
+        """
+        interval: float = 0.01
+        velocity_deadzone: float = (
+            0.15  # Adjusting this value may result in better/worse performance
```

Comment:
Could you give more description here? Like does increasing the deadzone value improve performance or vice versa?